backend/logic/tags/tags.js

```diff
@@ -18,6 +25,102 @@ var getAllTags = async () => {
   return tags;
 };
 
+/**
+ * Adds a tag into the database.
+ * @author Steven Li
+ * @param {String} tag
+ * @returns true if successful
+ * @throws error if MySQL connection failed
+ *         false if insertion failed
+ *         invalid format tag if tag is null
+ */
+var addTag = async tag => {
+  let connection = await mysql.getNewConnection();
```

Comment:
Minor detail but I think you should get the connection after verifying the tag format